### PR TITLE
Fix vbox ver 4 detection

### PR DESCRIPTION
**Pull request by pauliusm**
Fix version detection so docker-machine would work with VirtualBox 6.

### CODE REVIEW

File: xhyve/xhyve.go [hunk 1]
```diff
@@ -414,7 +414,7 @@ func (d *Driver) PreCreateCheck() error {
 	if err != nil {
 		return fmt.Errorf("Error detecting VBox version: %s", err)
 	}
-	if !strings.HasPrefix(ver, "5") {
+	if strings.HasPrefix(ver, "4") {
```
Comment (praveenkumar) [inline xhyve/xhyve.go]: 👍

**Comment by LeoUfimtsev** [comment]
I ran into this issue today (I'm running Virtualbox 6). Would be nice to see it merged.

**Comment by pauliusm** [comment]
@praveenkumar CI failure is not connected with my patch, it's most probably problem with ocaml packages:

```
# Run eval $(opam env) to update the current shell environment
The following dependencies couldn't be met:
  - io-page → ocaml < 4.06.0
      base of this switch (use `--unlock-base' to force)
No solution found, exiting
The command "opam --version && opam init && opam install uri qcow-format io-page.1.6.1 ocamlfind conf-libev" failed and exited with 20 during .
Your build has been stopped.
```

**Comment by LeoUfimtsev** [comment]
Thank you for merge. Do you guys have any idea in what version of minishift this fix would be available?

File: xhyve/xhyve.go [hunk 2]
```diff
@@ -414,7 +414,7 @@ func (d *Driver) PreCreateCheck() error {
...
 		return fmt.Errorf("Virtual Box version 4 or lower will cause a kernel panic" +
 			"if xhyve tries to run. You are running version: " +
```
Comment (dhergert) [inline xhyve/xhyve.go]: Gah, shoulda fixed the spacing typo between "panic" and "if" while you were in there!

**Comment by dhergert** [comment]
This isn't a minishift issue, it's a xhyve issue.  Minishift doesn't ship with xhyve, you probably downloaded and installed it with brew, ie
`brew install docker-machine-driver-xhyve`

My version shows as "0.3.3_1" which was released 2 years ago.  I guess the question for @praveenkumar is when do you think a new xhyve release will be cut with this fix?  Virtualbox 5.2 goes unsupported in 18 months and 6.0 has been out for quite some time now.

Thanks!

File: xhyve/xhyve.go (hunk 2 above)
Comment (guillaumegiordana) [reply xhyve/xhyve.go]: FYI : Doesn't work for version 6.0.4.

Error with pre-create check: "Virtual Box version 4 or lower will cause a kernel panicif xhyve tries to run. You are running version: 6.0.4r128413\n\n\t Please upgrade to version 5 at https://www.virtualbox.org/wiki/Downloads"

**Comment by kalbasit** [comment]
@praveenkumar this is impacting a lot of people, can you please cut a release?

**Comment by yackermann** [comment]
@praveenkumar Same thing here. When are you guys planing to release?

**Comment by kalbasit** [comment]
@zchee @SvenDowideit @pecigonzalo @LalatenduMohanty @gbraad Please either cut a release soon or mark this project as dead so we can fork it and move on. It's very frustrating to keep having to ask people to uninstall VirtualBox on their Macs.

**Comment by zchee** [comment]
@kalbasit sorry for late. will cut new release version asap

**Comment by kalbasit** [comment]
:tada: 

@zchee Thank you! You'll probably want to also close out #220 in that case! :beers:

**Comment by zchee** [comment]
@kalbasit 
Released `v0.4.0`
https://github.com/machine-drivers/docker-machine-driver-xhyve/releases/tag/v0.4.0

**Comment by zchee** [comment]
@kalbasit
Thanks for mentioned, just closed with comment 🙇‍♂️

**Comment by yackermann** [comment]
Ref https://github.com/Homebrew/homebrew-core/pull/39939

**Comment by zchee** [comment]
@herrjemand also Thanks :)

**Comment by kalbasit** [comment]
@zchee many many thanks!!
@herrjemand wow, that was fast! Thank you very much for that @chenrui333!

**Comment by IDCS1426** [comment]
You fix here, you break there. Using 0.4.0 lead to 

`Error starting the VM: Error creating new host: json: cannot unmarshal bool into Go struct field Driver.Virtio9p of type []string`

You use 0.3.3 you get 

`Error creating the VM. Error with pre-create check: Virtual Box version 4 or lower will cause a kernel panicif xhyve tries to run. You are running version: 6`

**Comment by condescendent** [comment]
> You fix here, you break there. Using 0.4.0 lead to
> 
> `Error starting the VM: Error creating new host: json: cannot unmarshal bool into Go struct field Driver.Virtio9p of type []string`
> 
> You use 0.3.3 you get
> 
> `Error creating the VM. Error with pre-create check: Virtual Box version 4 or lower will cause a kernel panicif xhyve tries to run. You are running version: 6`

HI, guys, I have gotten the same dilemma and have spent hours to get a solution. Could anyone help me with this ?  I am using 0.4.0 and would like to get a fix for this newest version. Anyone can kindly help me with this ?

**Comment by IDCS1426** [comment]
@condescendent Got it work on MacOS using xhyve

File: xhyve/xhyve.go (hunk 1 above)
Comment (djungowski) [reply xhyve/xhyve.go:417]: So, what you fix means, is:
- If a someone uses a Virtualbox Version lower than 4, there will be no error message
- If at some point we are t Virtualbox 40, the check will fail

**Pull request merged**